### PR TITLE
Package libpgport

### PR DESCRIPTION
**Pull request by syoliver**
Co-authored-by: Uilian Ries <uilianries@gmail.com>

Specify library name and version:  **libpq/12.2**

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/wiki#how-to-submit-a-pull-request) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/libpq/all/conanfile.py [hunk 1]
```diff
@@ -113,19 +114,33 @@ def build(self):
...
                         self.run("perl build.pl libpq")
+                        if not self.options.shared:
+                            self.run("perl build.pl libpgport")
         else:
             autotools = self._configure_autotools()
             with tools.chdir(os.path.join(self._source_subfolder, "src", "backend")):
                 autotools.make(target="generated-headers")
             with tools.chdir(os.path.join(self._source_subfolder, "src", "common")):
                 autotools.make()
+            if tools.Version(self.version) >= '12':
```
Comment (madebr) [inline recipes/libpq/all/conanfile.py:125]:
```suggestion
            if tools.Version(self.version) >= "12":
```

File: recipes/libpq/all/conanfile.py [hunk 2]
```diff
@@ -148,6 +165,12 @@ def package(self):
                 autotools.install()
             with tools.chdir(os.path.join(self._source_subfolder, "src", "interfaces", "libpq")):
                 autotools.install()
+            if tools.Version(self.version) >= '12':
```
Comment (madebr) [inline recipes/libpq/all/conanfile.py:168]:
```suggestion
            if tools.Version(self.version) >= "12":
```

File: recipes/libpq/all/test_package/conanfile.py [hunk 3]
```diff
@@ -8,6 +8,7 @@ class TestPackageConan(ConanFile):
 
     def build(self):
         cmake = CMake(self)
+        cmake.verbose = True
```
Comment (madebr) [inline recipes/libpq/all/test_package/conanfile.py:11]:
```suggestion
```

File: recipes/libpq/all/conanfile.py [hunk 4]
```diff
@@ -156,12 +179,44 @@ def package(self):
...
+            return "lib{}".format(name)
+        return  name
+
     def package_info(self):
         self.cpp_info.names["cmake_find_package"] = "PostgreSQL"
         self.cpp_info.names["cmake_find_package_multi"] = "PostgreSQL"
         self.env_info.PostgreSQL_ROOT = self.package_folder
-        self.cpp_info.libs = tools.collect_libs(self)
+        
+        self.cpp_info.components["pq"].libs = [self._construct_library_name("pq")]
```
Comment (prince-chrismc) [inline recipes/libpq/all/conanfile.py:192]: Components!!!

Comment (syoliver) [reply recipes/libpq/all/conanfile.py:192]: Yes thanks to components we can handle the mess of those libraries that depend of version, compiler dans link... 🤔

**Pull request merged**